hazelcast/src/main/java/com/hazelcast/nio/serialization/genericrecord/GenericRecordBuilder.java

```diff
@@ -97,14 +97,20 @@ static GenericRecordBuilder compact(String typeName) {
      *                  characters for {@link Portable} GenericRecords.
      * @param value     to set to GenericRecord
      * @return itself for chaining
-     * @throws HazelcastSerializationException if the field name does not exist
+     * @throws HazelcastSerializationException 1. if the field name does not exist
      *                                         in the schema/class definition or
      *                                         the type of the field does not
      *                                         match the one in the schema/class
      *                                         definition or the same field is
      *                                         trying to be set without using
      *                                         {@link
      *                                         GenericRecord#newBuilderWithClone()}.
+     *                                         2. if the field is set after the
+     *                                         GenericRecord is built. This can only
+     *                                         happen with a Compact GenericRecord
+     *                                         because Portable GenericRecords have a
+     *                                         ClassDefinition and they check for
+     *                                         field overwrites.
```

Comment:
I couldn't understand the last part. The following code fails currently with Portable generic records

```
        GenericRecordBuilder builder = GenericRecordBuilder.portable(
                new ClassDefinitionBuilder(1, 1)
                        .addIntField("foo")
                        .build()
                )
                .setInt32("foo", 5);

        builder.build();
        builder.setInt32("foo", 6);
```

Reply:
It fails but it is not due to setting a field after building. The error should say it is due to trying to override a field. So if you don't build it in the middle, it should fail again.

Reply:
ok to avoid confusion, let's remove the 

```
This can only
     *                                         happen with a Compact GenericRecord
     *                                         because Portable GenericRecords have a
     *                                         ClassDefinition and they check for
     *                                         field overwrites.
```

part